src/qt/statusbar.cpp

```diff
@@ -67,6 +65,9 @@ bool wxStatusBar::GetFieldRect(int i, wxRect& rect) const
     wxCHECK_MSG( (i >= 0) && ((size_t)i < m_panes.GetCount()), false,
                  "invalid statusbar field index" );
 
+    if(m_qtPanes->count() != m_panes.GetCount())
+        const_cast<wxStatusBar*>(this)->UpdateFields();
```

Comment:
Actually this doesn't look right; it looks more like it is hiding an error elsewhere.